corehq/apps/fixtures/views.py

```diff
@@ -575,3 +573,83 @@ def fixture_metadata(request, domain):
+
+    @property
+    def page_context(self):
+        return {
+            'save_url': reverse('csql_fixture_configuration', args=[self.domain]),
+            'csql_fixture_configurations':
+                list(self.all_module_badge_configurations().values('id', 'name', 'csql')),
+        }
+
+    def _post_response(self, message, div_class):
+        return HttpResponse((f'<div class="alert {div_class}">' + _(message) + '</div>'),
+                            content_type='text/html')
+
+    @atomic
+    def post(self, request, *args, **kwargs):
+        try:
+            data = request.POST
+            ''' Post data format is:
+                {
+                    'name': ['name1', 'name2', 'name3'],
+                    'id': ['1', '2', ''],  # empty string ID means new expression, missing means delete
+                    'csql': ['asdf', 'asdfg', 'asdfgh'],}
+                }
+            '''
+
+            touched_badge_ids = []
+            ids_list = data.getlist('id')
+            name_list = data.getlist('name')
+            csql_list = data.getlist('csql')
+
+            name_list_without_blanks = [name for name in name_list if name]
+            if len(name_list_without_blanks) != len(set(name_list_without_blanks)):
+                return self._post_response(
+                    "Configuration not updated, two expressions cannot have the same name.", 'alert-warning')
+
+            for index in range(0, len(ids_list)):
+                _id = ids_list[index]
+                name = name_list[index]
+                csql = csql_list[index]
+
+                if not name_list[index] or not csql_list[index]:
+                    if not name_list[index] and not csql_list[index]:
+                        continue  # ignore empty rows
+                    return self._post_response("Configuration not updated, some fields are blank.",
+                                               'alert-warning')
+
+                if _id:
+                    module_badge_configuration = CSQLFixtureExpression.by_domain(self.domain).get(id=_id)
+                    module_badge_configuration.name = name
+                    module_badge_configuration.csql = csql
+                    module_badge_configuration.save()
+                else:
+                    module_badge_configuration = CSQLFixtureExpression.objects.create(
+                        domain=self.domain, name=name, csql=csql)
+                touched_badge_ids.append(module_badge_configuration.id)
+            for expression in CSQLFixtureExpression.by_domain(self.domain).exclude(id__in=touched_badge_ids):
+                expression.soft_delete()
+            return self._post_response("Fixture confugration updated!", 'alert-success')
+        except Exception as e:
+            notify_exception(request, message=str(e))
+            return self._post_response("Configuration not updated, unknown error occurred.", 'alert-danger')
```

Comment:
What does HTMX do if this just 500s?

Reply:
that's lame - seems like you shouldn't need to build your own 500 handling in each case... Might have to figure out an easier pattern at some point.